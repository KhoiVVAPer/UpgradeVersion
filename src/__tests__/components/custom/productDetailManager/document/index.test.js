import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Document from '../../../../../components/custom/productDetailManager/document';

const setUp = (props) => {
  const component = shallow(<Document {...props} />);
  return component;
};

describe('components > custom > layoutManager > Document', () => {
  let component;
  const props = {
    layoutData: []
  };
  beforeEach(() => {
    component = setUp(props);
    return component;
  });

  it('Check snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders heading', () => {
    const input = component.find('SectionHeader').length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Check if renders FlatList for data', () => {
    const input = component.find('FlatList').length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Check if FlatList has renderItem props', () => {
    const input = component.find('FlatList').prop('renderItem');
    expect(input).toBeDefined();
  });

  it('Check if FlatList has scroll disabled', () => {
    const input = component.find('FlatList').prop('scrollEnabled');
    expect(input).toBeFalsy();
  });

  it('Check if FlatList has data', () => {
    const input = component.find('FlatList').prop('data');
    expect(input).toBeDefined();
  });
});
