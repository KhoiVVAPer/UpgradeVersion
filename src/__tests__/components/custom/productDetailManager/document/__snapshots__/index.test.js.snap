// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > custom > layoutManager > Document Check snapshot 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        <react-native-mock>
          <react-native-mock>
            DOWNLOADS
          </react-native-mock>
        </react-native-mock>
      </react-native-mock>
    </react-native-mock>
  </react-native-mock>
  <react-native-mock>
    <react-native-mock />
    <react-native-mock />
    <react-native-mock />
    <react-native-mock />
    <react-native-mock />
    <react-native-mock />
  </react-native-mock>
</react-native-mock>
`;
